coderd/workspaces.go

```diff
@@ -43,6 +43,19 @@ var (
 	errDeadlineBeforeStart = xerrors.New("new deadline must be before workspace start time")
 )
 
+// @Summary Get workspace metadata
+// @ID get-workspace
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Workspaces
+// @Param id path string true "Workspace ID"
+// @Param include_deleted query string false "Include deleted"
+// @Success 200 {object} codersdk.Workspace
+// @Failure 400 {object} codersdk.Response
+// @Failure 404 {object} codersdk.Response
+// @Failure 410 {object} codersdk.Response
+// @Failure 500 {object} codersdk.Response
```

Comment:
Do we need to be this thorough? Can't we just tell swaggo or swagger to just copy these to every route? Over the course of many API endpoints these really add up, and IMO they don't add much (especially 500, and other errors that may arrive from proxies etc.)

Reply:
I think that I can place shared entries directly to the [template file](https://github.com/mtojek/coder/blob/c090ef871e305d7758cae80594e9508eca1ff597/scripts/apidocgen/markdown-template/responses.def#L18), but AFAICS that would be only HTTP 500? Maybe HTTP 400 too. I don't know yet how common is the HTTP 404.

Alternatively, I can skip HTTP 500 and leave only the relevant ones like 400, 404, 410, etc.

Reply:
I wonder if it's even worth writing these errors down since they will change pretty often (especially as we add/change middleware).

Reply:
IMHO the more accurate, the better. If it's a hard problem to keep track of all middleware HTTP statuses, let's just track statuses relevant to particular calls. In theory, we should also include HTTP 403 as it may also appear.

I don't have a preference here, happy to hear more voices.

Reply:
My thinking is more that specifying the error codes is less accurate because we give the false impression that the list is complete (when it will most likely become out of sync in the future). If we instead say "if you don't get one of the success codes listed in the route then you will receive a `codersdk.Response` object then it's more accurate IMO. But yeah willing to see what others think too

Thoughts? @mafredri @coadler @johnstcn @Emyrk @f0ssel @kylecarbs

Reply:
Personally I think I'd rather specify just the success for the same reasons dean mentioned, it'll become out of date really easy unless we have a automatic test or linter for it. Also because any non-2XX should be a `codersdk.Response` it's should be easy to handle them all the same way.

Reply:
Dean has a good point. Some of these status codes are also pretty self explanatory. Like a 404, `Forbidden`, etc.

I think there is value in this if someone is generating code from it, but we are not. 

It would be best to have some way to document our site wide expected error codes and what they mean. Eg a 404 can really be a `Forbidden`, we just don't leak information using a 404

Reply:
👍 

I removed other responses than HTTP 200 for now. I can bring them back if necessary.